lib/KCard/index.vue

```diff
@@ -0,0 +1,226 @@
+                    scaleType: thumbnailScaleType,
+                  }"
+                  style="border-radius:13px;"
+                  class="spacing"
+                />
+                <slot v-if="!thumbnailSrc" name="thumbnailPlaceholder"></slot>
+              </aside>
+
+            </KGridItem>
+          </KGrid>
+
+          <footer class="spacing">
+            <slot name="footer"></slot>
+          </footer>
+        </div>
+      </div>
+    </template>
+  </BaseCard>
+
+</template>
+
+
+<script>
+
+  import KImg from '../KImg';
+  import BaseCard from './BaseCard.vue';
+
+  export default {
+    name: 'KCard',
+    components: { BaseCard, KImg },
+    props: {
+      title: {
+        type: String,
+        required: false,
+        default: null,
+      },
+      /**
+       * Sets the HTML heading level (h1, h2, etc.) for the title.
+       */
+      headingLevel: {
+        type: Number,
+        required: true,
+      },
+      /**
+       * The number of lines to display for the title. Defaults to 2.
+       */
+      titleLines: {
+        type: Number,
+        required: true,
+        default: 2,
+      },
+      /**
+       * An object containing the route definition for the link.
+       */
+      to: {
+        type: Object,
+        required: true,
+      },
+      /**
+       * The layout style of the hero banner. Required and cannot be empty.
+       *
+       * @validator
+       * @param {String} value - The layout value.
+       * @returns {Boolean} - True if the value is not empty, false otherwise.
+       */
+      layout: {
+        type: String,
+        required: true,
+        validator(value) {
+          if (!value) {
+            console.error('Error: Prop layout is required and can not be empty.');
+            return false;
+          }
+          return true;
+        },
+      },
+      /**
+       * Controls the display of the thumbnail image.
+       * Options: 'none' (default), 'block', or 'inline'.
+       */
+      thumbnailDisplay: {
+        type: String,
+        required: false,
+        default: 'none',
+      },
+      /**
+       * Set the source URL of the thumbnail image.
+       */
+      thumbnailSrc: {
+        type: String,
+        required: false,
+        default: null,
+      },
+      /**
+       * sets the scale type for the thumbnail image.
+       *  Options: 'centerInside' (default), 'cover', or 'contain'.
+       */
+      thumbnailScaleType: {
+        type: String,
+        required: false,
+        default: 'centerInside',
+      },
+    },
+    computed: {
+      KImgVerticalSmallStyle() {
+        if (this.thumbnailDisplay === 'small') {
+          return {
+            padding: '1em',
+          };
+        } else {
+          return {
+            padding: '0em',
+          };
+        }
+      },
+      isVerticalLayout() {
+        return (
+          this.layout === 'vertical' ||
+          (this.layout === 'horizontal' && this.thumbnailDisplay === 'large')
```

Comment:
Does this means that an horizontal layout with large thumbnail behaves like a vertical layout?

Reply:
Yes, my reasoning behind this  is that a large thumbnail in a horizontal layout might visually occupy a significant portion of the space, creating a layout that resembles a vertical one where elements are stacked.